modules/dynamic-operator/api/v1alpha1/kuberlogicservice_types.go

```diff
@@ -29,6 +30,23 @@ type KuberLogicServiceStatus struct {
 	Conditions []metav1.Condition `json:"conditions"`
 }
 
+type KuberLogicServiceSpec struct {
+	// Type of the cluster
+	// +kubebuilder:validation:Enum=postgresql;mysql;redis
+	Type string `json:"type"`
+	// Amount of replicas
+	// +kubebuilder:validation:Maximum=5
+	Replicas int32 `json:"replicas"`
+	// Volume size
+	VolumeSize string `json:"volumeSize,omitempty"`
+	// 2 or 3 digits: 5 or 5.7 or 5.7.31
+	// +kubebuilder:validation:Pattern=^\d+[\.\d+]*$
+	Version string `json:"version,omitempty"`
+
+	// any advanced configuration is supported
+	Advanced v11.JSON `json:"advanced,omitempty"`
```

Comment:
',inline' does not work correctly, need to implement custom converter from json. It's not ready yet.